plugins/hiscore/init.lua

```diff
@@ -158,8 +174,17 @@ function hiscore.startplugin()
 	  local r;
 	  if emu.softname() ~= "" then
 		local soft = emu.softname():match("([^:]*)$")
-		r = hiscore_path .. '/' .. emu.romname() .. "_" .. soft .. ".hi";
+		r = hiscore_path .. '/' .. emu.romname() .. '/' .. soft .. ".hi";
```

Comment:
Why the change in format here?  This also doesn’t account for the fact that there are three degrees of freedom: machine, list and software.  There are machines with multiple software lists with software items with identical short names in each list.  The `.hi` file format doesn’t seem to account for this either.

Reply:
can you make me an example or point me to some datfiles?

Reply:
I don't think there are any .hi files for it, but the Apple II machines (hash/apple2_flop*.xml) have multiple software lists with the same shortname, generally "original" (with copy protection intact), "clean cracked" (new modern-day crack with no defacement or removed screens/features), and sometimes just "cracked" (back-in-the-day crack with crack intros etc).

Reply:
`flop` devices are not supported currently, btw what `emu.romname()` returns in this case?

Reply:
@eadmaster Can you patch Johnny Nero: Action Hero MAME by bypassing the unencrypted data so we can re-dump it to be working and playable? (suicide battery means helps the key stored into FPGA on VP101 hardware like the Capcom Power System II hardware that known for Marvel vs. Capcom arcade game). ![s-l1600](https://user-images.githubusercontent.com/69607409/90078549-186c4d00-dcba-11ea-804d-b33dde54d970.jpg)

Reply:
@cuavas Johnny Nero: Action Hero will be playable on MAME 0.224 after patching the unencrypted data on the PCB with hard drive and redumping it.
![original (4)](https://user-images.githubusercontent.com/69607409/90078762-bd872580-dcba-11ea-8e9e-886571a811d2.jpg)